src/Model/Post.php

```diff
@@ -0,0 +1,390 @@
+				},
+				'pinged'        => function () {
+					return ! empty( $this->post->pinged ) && is_array( $this->post->pinged ) ? implode( ',', (array) $this->post->pinged ) : null;
+				},
+				'modified'      => function () {
+					return ! empty( $this->post->post_modified ) && '0000-00-00 00:00:00' !== $this->post->post_modified ? $this->post->post_modified : null;
+				},
+				'modifiedGmt'   => function () {
+					return ! empty( $this->post->post_modified_gmt ) ? Types::prepare_date_response( $this->post->post_modified_gmt ) : null;
+				},
+				'parent'        => function () {
+					$parent_post = ! empty( $this->post->post_parent ) ? get_post( $this->post->post_parent ) : null;
+					return isset( $parent_post->ID ) && isset( $parent_post->post_type ) ? DataSource::resolve_post_object( $parent_post->ID, $parent_post->post_type ) : $parent_post;
+				},
+				'editLast'      => function () {
+					$edit_last = get_post_meta( $this->post->ID, '_edit_last', true );
+					return ! empty( $edit_last ) ? DataSource::resolve_user( absint( $edit_last ) ) : null;
+				},
+				'editLock'      => function () {
+					$edit_lock       = get_post_meta( $this->post->ID, '_edit_lock', true );
+					$edit_lock_parts = explode( ':', $edit_lock );
+					return ! empty( $edit_lock_parts ) ? $edit_lock_parts : null;
+				},
+				'enclosure'     => function () {
+					$enclosure = get_post_meta( $this->post->ID, 'enclosure', true );
+					return ! empty( $enclosure ) ? $enclosure : null;
+				},
+				'guid'          => function () {
+					return ! empty( $this->post->guid ) ? $this->post->guid : null;
+				},
+				'menuOrder'     => function () {
+					return ! empty( $this->post->menu_order ) ? absint( $this->post->menu_order ) : null;
+				},
+				'link'          => function () {
+					$link = get_permalink( $this->post->ID );
+					return ! empty( $link ) ? $link : null;
+				},
+				'uri'           => function () {
+					$uri = get_page_uri( $this->post->ID );
+					return ! empty( $uri ) ? $uri : null;
+				},
+				'commentCount'  => function () {
+					return ! empty( $this->post->comment_count ) ? absint( $this->post->comment_count ) : null;
+				},
+				'featuredImage' => function () {
+					$thumbnail_id = get_post_thumbnail_id( $this->post->ID );
+					return ! empty( $thumbnail_id ) ? DataSource::resolve_post_object( $thumbnail_id, 'attachment' ) : null;
+				},
+			];
+
+			if ( 'attachment' === $this->post->post_type ) {
+				$attachment_fields = [
+					'caption' => function() {
+						$caption = apply_filters( 'the_excerpt', apply_filters( 'get_the_excerpt', $this->post->post_excerpt, $this->post ) );
+						return ! empty( $caption ) ? $caption : null;
+					},
+					'altText' => function() {
+						return get_post_meta( $this->post->ID, '_wp_attachment_image_alt', true );
+					},
+					'description' => function() {
+						return ! empty( $this->post->post_content ) ? apply_filters( 'the_content', $this->post->post_content ) : null;
```

Comment:
So the REST API outputs the caption as `RAW` and `RENDERED`. . .much like we do for the content of a Post. We should follow suit

And along with that only expose `RAW` to authenticated users.

Reply:
Agreed

Reply:
Now, the question is do we use the condition, (is_logged_in()) to apply here in the callback, or apply to the allowed_restricted_fields?

Feel like the latter is more centrally managed and plugins, etc can more easily declare whether the fields should/should not be exposed by filtering the `allowed_restricted_fields` array vs having to filter individual resolvers 🤔